docs/apm/api.asciidoc

```diff
@@ -274,6 +279,116 @@ POST /api/apm/settings/agent-configuration/search
 }
 --------------------------------------------------
 
+////
+*******************************************************
+*******************************************************
+////
+
+[role="xpack"]
+[[apm-annotation-api]]
+=== Annotation API
+
+The Annotation API allows you to annotate visualizations in the APM app with significant events, like deployments,
+allowing you to easily see how these events are impacting the performance of your existing applications.
+
+The following APIs are available:
+
+* <<apm-annotation-create>> to create an annotation for APM.
+// * <<obs-annotation-create>> POST /api/observability/annotation
+// * <<obs-annotation-get>> GET /api/observability/annotation/:id
+// * <<obs-annotation-delete>> DELETE /api/observability/annotation/:id
+
+By default, annotations are stored in a newly created `observability-annotations` index.
+The name of this index is configurable with `CONFIG_NAME_AND_LINK_HERE`.
+
 ////
 *******************************************************
 ////
+
+[[apm-annotation-config]]
+==== Create or update annotation
+
+[[apm-annotation-config-req]]
+===== Request
+
+`POST /api/apm/services/:serviceName/annotation`
+
+[role="child_attributes"]
+[[apm-annotation-config-req-body]]
+===== Request body
+
+`service`::
+(required, object) Service identifying the configuration to create or update.
++
+.Properties of `service`
+[%collapsible%open]
+======
+`version` :::
+  (required, string) Name of service.
+
+`environment` :::
+  (optional, string) Environment of service.
+======
+
+`@timestamp`::
+(required, string) The date and time of the annotation. Must be in https://www.w3.org/TR/NOTE-datetime[ISO 8601] format.
```

Comment:
ISO 8601? Any other formats listed on [this](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-date-format.html) page fail:
```
{
    "statusCode": 400,
    "error": "Bad Request",
    "message": "Invalid value \"1589165353320\" supplied to : ({ @timestamp: DateAsString, service: ({ version: string } & Partial<{ environment: string }>) } & Partial<{ message: string, tags: Array<string> }>)/0: { @timestamp: DateAsString, service: ({ version: string } & Partial<{ environment: string }>) }/@timestamp: DateAsString"
}
```

Reply:
yes: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString

That's not entirely consistent with how ES works, do you think that's okay?

Reply:
As long as it's documented, I think it's fine.